src/main/webapp/app/exercises/shared/structured-grading-criterion/grading-instructions-details/grading-instructions-details.component.ts

```diff
@@ -217,37 +213,33 @@ export class GradingInstructionsDetailsComponent implements OnInit, AfterContent
-     *          criteriaCommands, which consists of instructions that belong to a criterion
-     *       2. for each subarrray a method is called to create the criterion and instruction objects
-     * @param domainCommands containing tuples of [text, domainCommandIdentifiers]
+     * Creates criterion and instruction objects based on the parsed markdown text.
+     * @param textWithDomainActions The parsed text segments with their corresponding domain actions.
      */
-    createSubInstructionCommands(domainCommands: [string, DomainCommand | null][]): void {
-        let instructionCommands;
-        let criteriaCommands;
-        let endOfInstructionsCommand = 0;
-        if (!this.hasCriterionCommand(domainCommands)) {
-            this.setParentForInstructionsWithNoCriterion(domainCommands);
+    createSubInstructionActions(textWithDomainActions: TextWithDomainAction[]): void {
+        let instructionActions;
+        let criterionActions;
+        let endOfInstructionsAction = 0;
+        if (!this.hasCriterionAction(textWithDomainActions)) {
+            this.setParentForInstructionsWithNoCriterion(textWithDomainActions);
         } else {
-            for (const [, command] of domainCommands) {
-                endOfInstructionsCommand++;
-                this.setExerciseGradingInstructionText(domainCommands);
-                if (command instanceof GradingCriterionCommand) {
-                    instructionCommands = domainCommands.slice(0, endOfInstructionsCommand - 1);
-                    if (instructionCommands.length !== 0) {
-                        this.setParentForInstructionsWithNoCriterion(instructionCommands);
+            for (const { action } of textWithDomainActions) {
+                endOfInstructionsAction++;
+                this.setExerciseGradingInstructionText(textWithDomainActions);
+                if (action instanceof MonacoGradingCriterionAction) {
+                    instructionActions = textWithDomainActions.slice(0, endOfInstructionsAction - 1);
+                    if (instructionActions.length !== 0) {
+                        this.setParentForInstructionsWithNoCriterion(instructionActions);
                     }
-                    criteriaCommands = domainCommands.slice(endOfInstructionsCommand - 1);
-                    if (criteriaCommands.length !== 0) {
+                    criterionActions = textWithDomainActions.slice(endOfInstructionsAction - 1);
+                    if (criterionActions.length !== 0) {
                         this.instructions = []; // resets the instructions array to be filled with the instructions of the criteria
-                        this.groupInstructionsToCriteria(criteriaCommands); // creates criterion object for each criterion and their corresponding instruction objects
+                        this.groupInstructionsToCriteria(criterionActions); // creates criterion object for each criterion and their corresponding instruction objects
```

Comment:
I would either keep this comment or the one in line 221. Maybe merge them?

Reply:
I'll get rid of it 👍